logintoboggan.module

```diff
@@ -1046,20 +1046,20 @@ function logintoboggan_user_load($users) {
 /**
  * Implement hook_user_update().
  */
-function logintoboggan_user_update(&$edit, $account, $category) {
+function logintoboggan_user_update($account) {
   // Only perform this check if an admin is editing the account.
-  if (user_access('administer users') && isset($edit['roles'])) {
+  if (user_access('administer users') && isset($account->roles)) {
     // Check to see if roles present, and the pre-auth role was present when
     // the form was initially displayed.
-    if (isset($edit['logintoboggan_pre_auth_check'])) {
+    if (isset($account->logintoboggan_pre_auth_check)) {
```

Comment:
Here, we have a problem. The `$account` variable is a `User`object. This property `logintoboggan_pre_auth_check` does not exists in the user object, so this check will always return FALSE. 

In D7, this was part of the `$edit` array that this hook received. Now, since Backdrop doesn't have an `$edit` parameter, you'll need to attach this property to the user object itself during the form alter hook (`logintoboggan_form_user_profile_form_alter()`). So, in line 159, instead of 

```php
      $form['logintoboggan_pre_auth_check'] = array(
        '#type' => 'hidden',
        '#value' => '1',
      );
```

I would suggest:

```php
  $account->logintoboggan_pre_auth_check = TRUE
```

Please do check that this works.

Reply:
I fixed this as you noted and also found some other issues that were preventing this functionality from working and have fixed those. Thank you!